packages/block-library/src/template-part/edit/selection/index.js

```diff
@@ -3,15 +3,32 @@
  */
 import { __experimentalSearchForm as SearchForm } from '@wordpress/block-editor';
 import { useState } from '@wordpress/element';
+import { LEFT, RIGHT, UP, DOWN, BACKSPACE, ENTER } from '@wordpress/keycodes';
 /**
  * Internal dependencies
  */
 import TemplatePartPreviews from './template-part-previews';
 
+const preventArrowKeysPropagation = ( event ) => {
+	if (
+		[ LEFT, DOWN, RIGHT, UP, BACKSPACE, ENTER ].includes( event.keyCode )
+	) {
+		// Stop the key event from propagating up to ObserveTyping.startTypingInTextField.
+		event.stopPropagation();
+	}
+};
+const stopKeyPropagation = ( event ) => event.stopPropagation();
+
+// Disable reason (no-static-element-interactions): Navigational key-presses within
+// the menu are prevented from triggering WritingFlow and ObserveTyping interactions.
+/* eslint-disable jsx-a11y/no-static-element-interactions */
 export default function TemplatePartSelection( { setAttributes, onClose } ) {
 	const [ filterValue, setFilterValue ] = useState( '' );
 	return (
-		<>
+		<div
+			onKeyPress={ stopKeyPropagation }
```

Comment:
I'm curious why we need two different handlers here? Also curious, can we just attach these to `SearchForm`?

Reply:
> I'm curious why we need two different handlers here?

Im not entirely certain myself, but I saw that these were both used in the quick-inserter:
https://github.com/WordPress/gutenberg/blob/c2ea1b27df08117dbf53200630e8a627f4479df1/packages/block-editor/src/components/inserter/quick-inserter.js#L219-L220
My assumption is there are other controls that may fire off keyPress as well that we don't want to trigger from here.

> Also curious, can we just attach these to SearchForm?

Ah yeah, I added it to the wrapper for both before using composite to add controls for the list.  However it doesn't seem to stop the events when put on SearchForm, so Im guessing that component doesn't properly merge/handle these event handlers that are passed down to it.

Reply:
>  My assumption is there are other controls that may fire off keyPress as well that we don't want to trigger from here.

I guess I was wondering more why the two callbacks do slightly different things :)